opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -743,6 +781,100 @@ def set_up_accelerate_config(
+        embedding_dimension: int = 768,
+            Optional, the embedding_dimension of the model. If None, parse embedding_dimension from the config file of
+             pre-trained hugging-face model, if not found, default to be 768.
+        all_config: dict = None,
+            Optional, the all_config of the model. If None, parse all contents from the config file of pre-trained
+            hugging-face model.
+        model_type: str = None,
+            Optional, the model_type of the model. If None, parse model_type from the config file of pre-trained
+            hugging-face model.
+
+        Returns
+        -------
+        None
+        """
+        folder_path = self.folder_path
+        config_json_file_path = os.path.join(folder_path, "config.json")
+        if model_name is None:
+            model_name = self.model_id.split("/")[-1]
+
+        if all_config is None or model_type is None or embedding_dimension == 768:
+            if not os.path.exists(config_json_file_path):
+                raise Exception(
+                    str(
+                        "Cannot find config.json in"
+                        + config_json_file_path
+                        + ". Please check the config.son file in the path."
+                    )
+                )
+            f = open(config_json_file_path)
+            if all_config is None:
+                all_config = json.load(f)
+            if model_type is None:
+                if "model_type" in all_config.keys():
+                    model_type = all_config["model_type"]
+                else:
+                    raise Exception(
+                        str(
+                            "Cannot find model_type in config.json file"
+                            + config_json_file_path
+                            + ". Please check the config.son file in the path."
+                        )
+                    )
+
+            embedding_dimension_mapping_list = ["dim", "hidden_size", "d_model"]
+            for mapping_item in embedding_dimension_mapping_list:
+                if mapping_item in all_config.keys():
+                    embedding_dimension = all_config[mapping_item]
+                    break
+
+            f.close()
+
+        print("reading config file: at" + config_json_file_path)
+
+        default_file = {
+            "name": model_name,
+            "version": version_number,
+            "model_format": "TORCH_SCRIPT",
+            "model_task_type": "TEXT_EMBEDDING",
+            "model_config": {
+                "model_type": model_type,
+                "embedding_dimension": embedding_dimension,
+                "framework_type": "sentence_transformers",
+                "all_config": str(all_config),
+            },
+        }
+
+        print("generating model_config.json file...")
+        print(default_file)
+
+        model_config_file_path = os.path.join(folder_path, "model_config.json")
```

Comment:
line 818 using `config.json` , 
```
config_json_file_path = os.path.join(folder_path, "config.json")
```

Why use a different file name `model_config.json` here?

Suggest create constants for config file

Reply:
when we save the pre-trained sentence transformer model from hugging-face, we save the config file along in the same default folder, so I made the config file a bit different to prevent overwritten. 
for example, in the default file folder, which was initiated in the constructor, there are two config files. 
1. config.json  ---- this is file we save from pre-trained model 'sentence-transformers/msmarco-distilbert-base-tas-b'
2. model_config.json -- this is the file we generated for openSearch upload config. 

This is the reason, I hope it this make sense. If you come up with a better naming convention, we can change it to more meaningful file name.

Reply:
Got it, maybe we should make the name more readable, how about changing `config.json` -> `huggingface_config.json`, `model_config.json` -> `ml-commons_model_config.json`?

Reply:
sounds good for `model_config.json` -> `ml-commons_model_config.json`. It does make more sense that this config file is configuring for `ml-commons`

Reply:
for the huggingface one, just want to clarify,  the`config.json` file is from [sentence transformer ](https://github.com/UKPLab/sentence-transformers/blob/b86eec31cf0a102ad786ba1ff31bfeb4998d3ca5/sentence_transformers/SentenceTransformer.py#L339 ) package, after save, I can write another piece of code to rename the file, but that was the file name for all sentensce transformer config file. shall we keep it consistence? let me know if you prefer renaming it.

Reply:
I'm find to keep huggingface config file name as is. so no need to rename it.